packages/cli/src/services/types.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SlashCommand } from '../ui/commands/types.js';
+
+/**
+ * Defines the contract for any class that can load and provide slash commands.
+ * This allows the CommandService to be extended with new command sources
+ * (e.g., file-based, remote APIs) without modification.
+ *
+ * Loaders should receive any necessary dependencies (like Config) via their
+ * constructor.
+ */
+export interface ICommandLoader {
+  /**
+   * Discovers and returns a list of slash commands from the loader's source.
+   * @returns A promise that resolves to an array of SlashCommand objects.
+   */
+  loadCommands(): Promise<SlashCommand[]>;
```

Comment:
Should this be cancellable? I.e. should this take in an abort signal?

That way caller can determine timeout criteria